quantile/tdigest/tdigest.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tdigest
+
+import (
+	"math"
+	"sort"
+)
+
+var (
+	positiveInfinity = math.Inf(1)
+	negativeInfinity = math.Inf(-1)
+	sentinelCentroid = Centroid{Mean: positiveInfinity, Weight: 0.0}
+)
+
+type centroidsByMeanAsc []Centroid
+
+func (c centroidsByMeanAsc) Len() int           { return len(c) }
+func (c centroidsByMeanAsc) Less(i, j int) bool { return c[i].Mean < c[j].Mean }
+func (c centroidsByMeanAsc) Swap(i, j int)      { c[i], c[j] = c[j], c[i] }
+
+type tDigest struct {
+	compression      float64       // compression factor
+	mergedCapacity   int           // merged centroid slice capacity
+	unmergedCapacity int           // unmerged centroid slice capacity
+	multiplier       int64         // quantile precision multiplier
+	centroidsPool    CentroidsPool // centroids pool
+
+	closed         bool       // whether the t-digest is closed
+	merged         []Centroid // merged centroids
+	mergedWeight   float64    // total weight of merged centroids
+	unmerged       []Centroid // unmerged centroid slice capacity
+	unmergedWeight float64    // total weight of unmerged centroids
+	minValue       float64    // minimum value
+	maxValue       float64    // maximum value
+}
+
+// mergedCapacity computes the capacity of the merged centroid slice
+func mergedCapacity(compression float64) int {
+	return int(math.Pi*compression/2 + 0.5)
+}
+
+func unmergedCapacity(compression float64) int {
+	compression = math.Min(math.Max(20, compression), 1000)
+	return (int)(7.5 + 0.37*compression - 2e-4*compression*compression)
```

Comment:
i.e. any reason not to use the more common cast that also uses less parentheses, e.g.:
```go
int(7.5 + 0.37*compression - 2e-4*compression*compression)
```

Rather than currently:
```go
(int)(7.5 + 0.37*compression - 2e-4*compression*compression)
```